apps/emqx/src/emqx_cm.erl

```diff
@@ -358,7 +352,7 @@ pick_channel(ClientId) ->
             ?SLOG(warning, #{msg => "more_than_one_channel_found", chan_pids => ChanPids}),
             lists:foreach(
                 fun(StalePid) ->
-                    catch discard_session(ClientId, StalePid)
+                    discard_session(ClientId, StalePid)
```

Comment:
Just some findings, while I am investigating some issue in the same area, I found one issue:

The computation of StalePids seems wrong in cluster setup or even on single node as well, we could not assume 'later' channel is the latest one as the inserts are now all dirty, meaning we might discard the wrong channel and take over the wrong session.

Reply:
Curious how you define _wrong_ here? Having more than one channel for a given ClientID is an anomaly, which in my opinion suggests that there's no good definition of _wrong_ or _right_ channel: e.g. imagine there was a brain-split, and 2 lingering sessions now hold disjoint sets of subscriptions / mqueues / etc.

So we either have to bring strict serializability to the table, or tolerate multiple entries and rely on some heuristic. Do you suggest using "pick the most recent one" heuristic, regardless of the potential clock skew?